k8s/sonarqube/README.md

```diff
@@ -0,0 +1,449 @@
+until you are ready to upgrade. To get the digest for the image, use the
+following script:
+
+```shell
+for i in "IMAGE_SONARQUBE" "IMAGE_POSTGRESQL" "IMAGE_POSTGRESQL_EXPORTER" "IMAGE_METRICS_EXPORTER"; do
+  repo=$(echo ${!i} | cut -d: -f1);
+  digest=$(docker pull ${!i} | sed -n -e 's/Digest: //p');
+  export $i="$repo@$digest";
+  env | grep $i;
+done
+```
+
+Create a certificate for PostgreSQL. If you already have a certificate that you
+want to use, copy your certificate and key pair in to the `server.crt` and
+`server.key` files.
+
+```shell
+# create a certificate for postgresql
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+    -keyout /tmp/server.key \
+    -out /tmp/server.crt \
+    -subj "/CN=postgresql/O=postgresql"
+
+kubectl --namespace $NAMESPACE create secret generic $APP_INSTANCE_NAME-tls \
+        --from-file=/tmp/server.crt --from-file=/tmp/server.key
+```
+
+Generate random password for PosgreSQL:
+
+```shell
+export POSTGRESQL_DB_PASSWORD=$(openssl rand 9 | openssl base64 -A | openssl base64)
+```
+
+Enable Stackdriver Metrics Exporter:
+
+> **NOTE:** Your GCP project should have Stackdriver enabled. For non-GCP clusters, export of metrics to Stackdriver is not supported yet.
+
+By default the integration is disabled. To enable, change the value to `true`.
+
+```shell
+export METRICS_EXPORTER_ENABLED=false
+```
+
+#### Expand the manifest template
+
+Use `helm template` to expand the template. We recommend that you save the
+expanded manifest file for future updates to the application.
+
+```shell
+helm template chart/sonarqube \
+  --name="$APP_INSTANCE_NAME" \
+  --namespace="$NAMESPACE" \
+  --set sonarqube.image=$IMAGE_SONARQUBE \
+  --set postgresql.image=$IMAGE_POSTGRESQL \
+  --set postgresql.exporter.image=$IMAGE_POSTGRESQL_EXPORTER \
+  --set postgresql.db.password=$POSTGRESQL_DB_PASSWORD  \
+  --set metrics.image=$METRICS_EXPORTER_ENABLED > ${APP_INSTANCE_NAME}_manifest.yaml
+```
+#### Apply the manifest to your Kubernetes cluster
+
+Use `kubectl` to apply the manifest to your Kubernetes cluster:
+
+```shell
+kubectl apply -f "${APP_INSTANCE_NAME}_manifest.yaml" --namespace "${NAMESPACE}"
+```
+
+#### View the app in the Google Cloud Console
+
+To get the GCP Console URL for your app, run the following command:
+
+```shell
+echo "https://console.cloud.google.com/kubernetes/application/${ZONE}/${CLUSTER}/${NAMESPACE}/${APP_INSTANCE_NAME}"
+```
+
+To view the app, open the URL in your browser.
+
+# Using the SonarQube Community Edition
+
+By default, the application is not exposed externally. To get access to SonarQube, run the following command:
+
+```bash
+kubectl port-forward --namespace $NAMESPACE svc/$APP_INSTANCE_NAME-sonarqube-svc 9000:9000
+```
+
+Then, navigate to the [http://localhost:9000/metrics](http://localhost:9000/metrics) endpoint. Use the username `admin` and password `admin` to login.
+
+To get access web-page with default credentials:
```

Comment:
Can you delete L224-229, as we already included credentials in the sentence above?

Reply:
Done